.github/workflows/ux-pages.yml

```diff
@@ -0,0 +1,55 @@
+name: Deploy UX to GitHub Pages
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["features/ux"]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets the GITHUB_TOKEN permissions to allow deployment to GitHub Pages
+permissions:
+    contents: read
+    pages: write
+    id-token: write
+  
+
+# Allow one concurrent deployment
+concurrency:
+    group: 'pages'
+    cancel-in-progress: true
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+      - name: Install dependencies
+        run: npm ci
+        working-directory: ./src/web
+      - name: Clean and Build
+        run: |
+          rm -rf ./dist || true
+          NODE_ENV=production npm run build
+        working-directory: ./src/web
+      - name: Setup Pages
+        uses: actions/configure-pages@v4
+      - name: List build output
+        run: ls -la ./src/web/dist
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          # Upload dist folder
+          path: './src/web/dist'
+      - name: Deploy to GitHub Pages
```

Comment:
Will this impact our current GitHub Pages site when this PR completes? We need to make sure not to touch that until we're done and ready to go live. Not sure if this needs to be commented out in the meantime. Or maybe we just make it only manually triggered?